src/runtime/typemanager.cs

```diff
@@ -817,4 +963,183 @@ internal static void CopySlot(IntPtr from, IntPtr to, int offset)
             Marshal.WriteIntPtr(to, offset, fp);
         }
     }
+
+    class SlotsHolder
+    {
+        public const string HolderKeyName = "_slots_holder";
+        public delegate void Resetor(IntPtr type, int offset);
```

Comment:
NIT: this delegate type looks like a generic action on a slot. Maybe `SlotAction` or `TypeSlotAction`?

Reply:
Yes, but the only action performed/context it is used for now is Reset.

Reply:
ResetSlotAction?

Some documentation of what the function should do + what the arguments mean would be nice.